### PR TITLE
Add memory module

### PR DESCRIPTION
**Pull request by tuxudo**
Created memory module.

### CODE REVIEW

File: app/modules/memory/memory_model.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+use CFPropertyList\CFPropertyList;
+
+class Memory_model extends \Model {
+
+	function __construct($serial='')
+	{
+		parent::__construct('id', 'memory'); //primary key, tablename
+		$this->rs['id'] = '';
+		$this->rs['serial_number'] = $serial;
+        $this->rs['name'] = '';
```
Comment (bochoven) [inline app/modules/memory/memory_model.php]: please fix indenting

File: app/modules/memory/memory_model.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		$this->rs['dimm_status'] = '';
+		$this->rs['dimm_manufacturer'] = '';
+		$this->rs['dimm_part_number'] = '';
+		$this->rs['dimm_serial_number'] = '';
+		$this->rs['dimm_ecc_errors'] = '';
+		$this->rs['global_ecc_state'] = 0; // can be 0, 1, or 2
+		$this->rs['is_memory_upgradeable'] = 0; // true/false
+
+		// Schema version, increment when creating a db migration
+		$this->schema_version = 0;
```
Comment (bochoven) [inline app/modules/memory/memory_model.php]: schema version is v2

File: app/modules/memory/memory_model.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+		// Add indexes
+		$this->idx[] = array('serial_number');
```
Comment (bochoven) [inline app/modules/memory/memory_model.php]: no need for indexes in v3

**Comment by tuxudo** [comment]
This PR isn't ready to merge yet, I am looking to expand its functionality

**Comment by ofirgalcon** [comment]
Works, no issues so far. Even works with VMs.

**Comment by tuxudo** [comment]
PR moved to module repo, closing.

**Pull request closed without merging**